packages/algorand/src/adapter.ts

```diff
@@ -0,0 +1,89 @@
+import { generateHashLock, getExpiration, sha256 } from '@jelly-swap/utils';
+import BigNumber from 'bignumber.js';
+
+import { JellyAdapter, ContractSwap, UserInputSwap } from './types';
+import Config from './config';
+
+export default class AlgoAdapter implements JellyAdapter {
+    private config: any;
+
+
+    constructor(config = Config()) {
+        this.config = config;
+    }
+
+    createSwapFromInput(inputSwap: ContractSwap, sender = this.config.receiverAddress, currentBlock?: number): ContractSwap {
```

Comment:
The correct interface should be:
```
createSwapFromInput(inputSwap, sender)
```

This additional parameter (`currentBlock`) will break the Open-closed Principle, enforcing the clients to handle that difference in the interface.

One possible solution would be to fetch the latest block inside the function.

Reply:
this will make the function async and in the other cases formatInput is not an async function - which will aslo break the interface.  

I am not sure that fetching the latest block is a job for the adapter. It should be a pure function IMO

Reply:
Reading the block number inside the adapter will not have any side effects (in terms of the final result), i.e. the output will remain the same whether you pass the current block as an argument or you fetch it inside the function. In that case the function will remain pure. Also the idea of the adapter is to handle all differences for every specific blockchain. In the Algorand case the difference is that it needs blocks instead of timestamp, so it either needs to be handled in the adapter or in the `newContract` function.